extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayCreator.java

```diff
@@ -33,7 +33,13 @@ public FlywayCreator withCallbacks(Collection<Callback> callbacks) {
 
     public Flyway createFlyway(DataSource dataSource) {
         FluentConfiguration configure = Flyway.configure();
-        configure.dataSource(dataSource);
+
+        if (flywayRuntimeConfig.username.isPresent() && flywayRuntimeConfig.password.isPresent()) {
+            configure.dataSource(
+                    new DataSourceDecorator(dataSource, flywayRuntimeConfig.username, flywayRuntimeConfig.password));
```

Comment:
You can invoke `get()` here and avoid using `Optional` in the class constructor

Reply:
@gsmet Before I read your comments on the previous PR, I suggested @mfpc to use this pattern. The idea was to reuse the DataSource configuration and call `DataSource#getConnection(String user, String password)` when `DataSource#getConnection()` is called in the Flyway codebase. 

But now I see that your suggestion seems nicer, so 👍🏻 to that